pkl-core/src/main/java/org/pkl/core/module/ModuleKeys.java

```diff
@@ -771,4 +784,151 @@ public boolean hasElement(SecurityManager securityManager, URI elementUri)
       return projectResolver.getResolvedDependenciesForPackage(packageUri, dependencyMetadata);
     }
   }
+
+  public static class External implements ModuleKey {
+    public static class Resolver {
```

Comment:
Let's move class `Resolver` to its own class, and use it both here and for the resource reader (see `FileResolver`, `PackageResolver`, etc).